modal/queue.py

```diff
@@ -309,9 +332,12 @@ async def _put_many_nonblocking(self, vs: List[Any]):
             raise queue.Full(exc.message) if exc.status == Status.RESOURCE_EXHAUSTED else exc
 
     @live_method
-    async def len(self) -> int:
-        """Return the number of objects in the queue."""
-        request = api_pb2.QueueLenRequest(queue_id=self.object_id)
+    async def len(self, partition: Optional[str] = None) -> int:
+        """Return the number of objects in the queue partition."""
```

Comment:
Haven't looked server-side, but what happens if you have a partitioned queue but don't specify `partition` here? Does it return the full queue size? Or is there some default "anonymous" partition? Or does it raise?

Reply:
Yeah good point, the default partition is the empty key. I disallow specifying an empty partition client side (it's either `None` or a non-zero string).

Reply:
Going to add some documentation in a follow up!